File: 0409-longest-palindrome/0409-longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        letters = [0 for i in range(52)]
        for letter in s:
            if letter.isupper():
                letters[ord(letter) - ord('A')] += 1
            else:
                letters[ord(letter) - ord('a') + 26] += 1
        odd = False
        result = 0
        for num in letters:
            if num % 2:
                if not odd:
                    odd = True
                    result += num
                else:
                    result += num - 1
            else:
                result += num
        return result

File: 0409-longest-palindrome/0409-longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        int[] letters = new int[52];
        for (int i = 0; i < s.length(); i++) {
            if (Character.isUpperCase(s.charAt(i))) {
                letters[s.charAt(i) - 'A']++;
            } else {
                letters[s.charAt(i) - 'a' + 26]++;
            }
        }
        boolean odd = false;
        int result = 0;
        for (int num: letters) {
            if (num % 2 == 1) {
                if (!odd) {
                    odd = true;
                    result += num;
                } else {
                    result += num - 1;
                }
            } else {
                result += num;
            }
        }
        return result;
    }
}